### ISSUE TITLE
Switch to using QCElemental `Molecule.identifiers` for CMILES and friends?

### ISSUE THREAD

**Issue by dotsdl** [openforcefield/openff-qcsubmit#199]
Historically we used `qcelemental.models.Molecule.extras` for storing e.g. `canonical_isomeric_explicit_hydrogen_mapped_smiles`, but we may wish to start using `qcelemental.models.Molecule.identifier` for this information in addition, and eventually, instead.

Any objections to this?

**Comment by jthorton** [comment]
@dotsdl good idea the identifiers [here](https://github.com/MolSSI/QCElemental/blob/d04a380fd80e6634648ae7fe4c4305dd1f4b5070/qcelemental/models/molecule.py#L76) seem to support most of our needs, it would be good if we could extend it though to support all of the attributes we currently use, as we also use the fixed hydrogen layer inchi and inchikey which allows us to distinguish between tautomers when performing searches on our results collections see all of our used attributes [here](https://github.com/openforcefield/openff-qcsubmit/blob/e4d5add8f169b7c358fb518358f12699923cee4d/openff/qcsubmit/common_structures.py#L810).

We also use the hill formula as our molecular formula it would be good to clarify what is supposed to be used in the identifier `molecular_formula` field as I see they support two types of [formula](https://github.com/MolSSI/QCElemental/blob/d04a380fd80e6634648ae7fe4c4305dd1f4b5070/qcelemental/models/molecule.py#L785) as I belive you can search qcarchive using this formula so it would be good to be consistent. 

It might also be good to suggest that the `molecule_hash` is generated during the init of the molecule as I think its supposed to be the hash from this [function](https://github.com/MolSSI/QCElemental/blob/d04a380fd80e6634648ae7fe4c4305dd1f4b5070/qcelemental/models/molecule.py#L760) but currently would allow any string to be used. 

Maybe we should also suggest a provenance field so we can supply the toolkit used and the version which made the identifiers.
